File: src/cpp/ch12_2_3_ex1/dynamic_binding_example.cpp
#include <iostream>
using namespace std;

class A {
public:
  virtual void draw() { cout << "Object A: " << this << endl; }
};

class B : public A {
public:
  void draw() { cout << "Object B: " << this << endl; }
};

int main() {
  A *r = new A();
  r->draw();
  r = new B();
  r->draw();
}

File: src/java/ch12_2_3_ex1/DynamicBindingExample.java
package ch12_2_3_ex1;

class A {
    public void draw() {
        System.out.println("Object A: " + this);
    }
}

class B extends A {
    public void draw() {
        System.out.println("Object B: " + this);
    }

}

public class DynamicBindingExample {
    public static void main(String[] args) {
        A ref = new A();
        ref.draw();
        ref = new B();
        ref.draw();
    }
}